### PR TITLE
[16.0] [MIG] delivery_carrier_warehouse

### PR DESCRIPTION
**Pull request by santostelmo**
delivery_carrier_warehouse Migration to 16.0

### CODE REVIEW

File: delivery_carrier_warehouse/tests/test_delivery_carrier_warehouse.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+class TestSaleDeliveryCarrierPreference(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+        ref = cls.env.ref
+        cls.partner = ref("base.res_partner_12")
+        cls.product = ref("product.product_product_20")
+        cls.free_delivery_carrier = ref("delivery.free_delivery_carrier")
+        cls.normal_delivery_carrier = ref("delivery.normal_delivery_carrier")
```
Comment (jbaudoux) [inline delivery_carrier_warehouse/tests/test_delivery_carrier_warehouse.py]:
```suggestion
        cls.local_delivery_carrier = ref("delivery.delivery_local_delivery")
```

File: delivery_carrier_warehouse/tests/test_delivery_carrier_warehouse.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        sale_form.partner_id = cls.partner
+        with sale_form.order_line.new() as line_form:
+            line_form.product_id = cls.product
+            line_form.product_uom_qty = 2
+        return sale_form.save()
+
+    def test_warehouse_carrier(self):
+        order = self._create_sale_order()
+        self.assertEqual(
+            self.partner.property_delivery_carrier_id, self.normal_delivery_carrier
```
Comment (jbaudoux) [inline delivery_carrier_warehouse/tests/test_delivery_carrier_warehouse.py]:
```suggestion
            self.partner.property_delivery_carrier_id, self.local_delivery_carrier
```

File: delivery_carrier_warehouse/tests/test_delivery_carrier_warehouse.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        )
+        action = order.action_open_delivery_wizard()
+        default_carrier_id = action["context"]["default_carrier_id"]
+        self.assertEqual(default_carrier_id, self.normal_delivery_carrier.id)
```
Comment (jbaudoux) [inline delivery_carrier_warehouse/tests/test_delivery_carrier_warehouse.py]:
```suggestion
        self.assertEqual(default_carrier_id, self.local_delivery_carrier.id)
```

**Review by jbaudoux — CHANGES_REQUESTED** [review]
xmlid has been renamed in delivery demo data

**Review by Ricardoalso — APPROVED** [review]
LGTM

**Review by jbaudoux — APPROVED** [review]

**Pull request merged**